lms/static/scripts/frontend_apps/components/SubmitGradeForm.tsx

```diff
@@ -139,12 +148,20 @@ export default function SubmitGradeForm({
     }
   };
 
-  const handleInput = (e: Event) => {
-    // If any input is detected, close the ValidationMessage.
-    setValidationError(false);
-    setGradeSaved(false);
-    setDraftGradeValue((e.target as HTMLInputElement).value);
-  };
+  const handleInput = useCallback(
+    (e: Event) => {
+      // If any input is detected, close the ValidationMessage.
+      setValidationError(false);
+      setGradeSaved(false);
+
+      const newDraftGradeValue = (e.target as HTMLInputElement).value;
+      setDraftGradeValue(newDraftGradeValue);
+
+      // Check if there are unsavedChanges
+      updateHasUnsavedChanges(newDraftGradeValue !== grade.data);
```

Comment:
Note to future self: When we add the grading comment, this callback will have to take it into consideration as well. Same for the comment itself, which will have to take the grade value into consideration.